File: plugins/swift-kit/templates/optional_bind.swift
// Named boundary: prefer if let / guard let / ?? over force unwrap.
// Copy into product sources; keep swift-rg-allow only on intentional seams.

func firstLabel(in values: [String?]) -> String? {
    guard let first = values.first else { return nil }
    return first ?? "missing"
}

func requireLabel(_ value: String?) throws -> String {
    guard let value else {
        throw NSError(domain: "swift-kit", code: 1)
    }
    return value
}

File: plugins/kotlin-kit/templates/optional_bind.kt
// Named boundary: prefer ?. / ?: / requireNotNull over !!.
// Copy into product sources; keep kotlin-rg-allow only on intentional seams.

fun firstLabel(values: List<String?>): String? {
    val first = values.firstOrNull() ?: return null
    return first
}

fun requireLabel(value: String?): String {
    return requireNotNull(value) { "label required" }
}
